modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/EditObjectFolderHeader/ModalPublishObjectDefinitions.tsx

```diff
@@ -0,0 +1,519 @@
+					updateObjectDefinitionStatus(prevState, id, status)
+				);
+
+				return publishObjectDefinition(id);
+			}
+		);
+
+		try {
+			const responses = await Promise.all(publishPromises);
+
+			const hasRejectedResponses = responses.some(
+				(response) =>
+					typeof response === 'number' && response === STATUS.REJECTED
+			);
+			const filteredResponses = responses.filter(
+				(response) => typeof response === 'object'
+			);
+
+			setModalHeaderMessage(
+				!hasRejectedResponses
+					? Liferay.Language.get('successfully-published')
+					: Liferay.Language.get('published-with-errors')
+			);
+			setPublishObjectDefinitionsStatus(
+				!hasRejectedResponses ? STATUS.APPROVED : STATUS.REJECTED
+			);
+
+			const newElements = elements.map((element) => {
+				if (isNode(element)) {
+					const elementId =
+						(element as FlowElement<ObjectDefinitionNodeData>).data
+							?.id || 0;
+
+					const response = (filteredResponses as ObjectDefinition[]).find(
+						(response) => response.id === elementId
+					);
+
+					if (response) {
+						return {
+							...element,
+							data: {
+								...element.data,
+								status: response.status,
+							},
+						};
+					}
+
+					return element;
+				}
+
+				return element;
+			});
+
+			dispatch({
+				payload: {
+					newElements,
+				},
+				type: TYPES.SET_ELEMENTS,
+			});
+		}
+		catch (error) {
+			setModalHeaderMessage(Liferay.Language.get('confirm-publishing'));
+			setPublishObjectDefinitionsStatus(STATUS.REJECTED);
+		}
+	};
+
+	const handleSelectAllObjectDefinitions = (
+		actionType?: 'checkAll' | 'checkRemoveAll'
+	): void => {
+		if (actionType) {
+			const allSelectedDraftObjectDefinitions =
+				selectedDraftObjectDefinitions.length ===
+				draftObjectDefinitionNodes.length;
+
+			if (
+				allSelectedDraftObjectDefinitions &&
+				actionType !== 'checkAll'
+			) {
+				setSelectAllDraftObjectDefinitions(false);
+				setSelectedDraftObjectDefinitions([]);
+			}
+			else {
+				const newSelectedDraftObjectDefinitions = draftObjectDefinitionNodes.map(
+					(draftObjectDefinitionNode) => {
+						const {data} = draftObjectDefinitionNode;
+
+						return {id: data?.id!, status: data?.status!};
+					}
+				);
+
+				setSelectAllDraftObjectDefinitions(true);
+				setSelectedDraftObjectDefinitions(
+					newSelectedDraftObjectDefinitions
+				);
+			}
+		}
+	};
+
+	const handleCheckboxChange = (objectDefinitionId: number): void => {
+		if (
+			selectedDraftObjectDefinitions.some(
+				(selectedDraftObjectDefinition) =>
+					selectedDraftObjectDefinition.id === objectDefinitionId
+			)
+		) {
+			setSelectedDraftObjectDefinitions(
+				selectedDraftObjectDefinitions.filter(
+					(selectedDraftObjectDefinition) =>
+						selectedDraftObjectDefinition.id !== objectDefinitionId
+				)
+			);
+		}
+		else {
+			const objectDefinition = objectDefinitionNodes.find(
+				(objectDefinitionNode) =>
+					objectDefinitionNode.data?.id === objectDefinitionId
+			)!;
```

Comment:
hey @naraandrade, the return of this find method will be an `objectDefinitionNode`. The name constant `objectDefinition` is not semantic here. So we can change to `objectDefinitionNode` and create a pattern to use in find current item to not repeat the names.

Reply:
Hey @naraandrade, I believe you forgot to make this changes.